server/src/main/java/org/opensearch/wlm/tracker/TaskResourceUsageCalculator.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.wlm.tracker;
+
+import org.opensearch.core.tasks.resourcetracker.ResourceStats;
+import org.opensearch.tasks.Task;
+import org.opensearch.wlm.ResourceType;
+
+import java.util.function.Supplier;
+
+import static org.opensearch.wlm.tracker.QueryGroupResourceUsageTrackerService.HEAP_SIZE_BYTES;
+
+/**
+ * Utility class to calculate task level resource usage
+ */
+public abstract class TaskResourceUsageCalculator {
```

Comment:
IMO, instead of making `TaskResourceUsageCalculator` abstract class, we can have it as interface and take implementation as input within the enum?

```
ResourceType(String name, TaskResourceUsageCalculator taskResourceUsageCalculator, boolean statsEnabled) 
```

Reply:
Doing this will make the enum consuming classes hard to test since we can't make static entities